packages/next/src/shared/lib/router/router.ts

```diff
@@ -497,7 +497,12 @@ function fetchNextData({
       headers: Object.assign(
         {} as HeadersInit,
         isPrefetch ? { purpose: 'prefetch' } : {},
-        isPrefetch && hasMiddleware ? { 'x-middleware-prefetch': '1' } : {}
+        isPrefetch && hasMiddleware
+          ? {
+              'x-middleware-prefetch': '1',
+              'x-deployment-id': process.env.NEXT_DEPLOYMENT_ID,
```

Comment:
I think we should just ensure this header is always passed for requests, even if they're not prefetch / middleware is present.

Reply:
Just bumping this, if you're able to make this change we could re-review!

Reply:
Thanks for bumping! Updated the PR 👍